rules/peer_restrictions_test.go

```diff
@@ -150,6 +150,49 @@ func TestPeerRestrictCheckRequest(t *testing.T) {
 		},
 	)
 	require.NoError(t, err)
+
+	// Test that we can open a channel to a non-restricted peer.
+	openReqNonRestricted := &lnrpc.OpenChannelRequest{
+		NodePubkey: peerKey3[:],
+	}
+	_, err = enf.HandleRequest(
+		ctx, "/lnrpc.Lightning/OpenChannelSync", openReqNonRestricted,
+	)
+	require.NoError(t, err)
+	_, err = enf.HandleRequest(
+		ctx, "/lnrpc.Lightning/OpenChannel", openReqNonRestricted,
+	)
+	require.NoError(t, err)
+
+	// Test that we deny channel opening for a restricted peer.
+	openReqRestricted := &lnrpc.OpenChannelRequest{
+		NodePubkey: peerKey1[:],
+	}
+	_, err = enf.HandleRequest(
+		ctx, "/lnrpc.Lightning/OpenChannelSync", openReqRestricted,
+	)
+	require.ErrorContainsf(t, err, "illegal action on peer in peer ", "")
+	_, err = enf.HandleRequest(
+		ctx, "/lnrpc.Lightning/OpenChannel", openReqRestricted,
+	)
+	require.ErrorContainsf(t, err, "illegal action on peer in peer ", "")
+
+	// Test that we cannot perform a batched channel opening when a
+	// restricted peer is present.
+	batchOpenReq := &lnrpc.BatchOpenChannelRequest{
+		Channels: []*lnrpc.BatchOpenChannel{
+			{
+				NodePubkey: peerKey1[:],
+			},
+			{
+				NodePubkey: peerKey3[:],
+			},
+		},
+	}
+	_, err = enf.HandleRequest(
+		ctx, "/lnrpc.Lightning/BatchOpenChannel", batchOpenReq,
+	)
+	require.ErrorContainsf(t, err, "illegal action on peer in peer ", "")
```

Comment:
I think it makes sense to include test coverage of a `BatchOpenChannelRequest` that actually does succeed.